pkg/util/mount/safe_format_and_mount_test.go

```diff
@@ -105,6 +105,44 @@ func TestSafeFormatAndMount(t *testing.T) {
 			},
 			expectedError: nil,
 		},
+		{ // Test a normal mount
+			fstype: "xfs",
+		},
+		{ // Test that 'lsblk' is called and fails
+			fstype:    "xfs",
+			mountErrs: []error{fmt.Errorf("unknown filesystem type '(null)'")},
+			execScripts: []ExecArgs{
+				{"lsblk", []string{"-nd", "-o", "FSTYPE", "/dev/foo"}, "xfs", nil},
+			},
+			expectedError: fmt.Errorf("unknown filesystem type '(null)'"),
+		},
+		{ // Test that 'lsblk' is called and confirms unformatted disk, format fails
+			fstype:    "xfs",
+			mountErrs: []error{fmt.Errorf("unknown filesystem type '(null)'")},
+			execScripts: []ExecArgs{
+				{"lsblk", []string{"-nd", "-o", "FSTYPE", "/dev/foo"}, "", nil},
+				{"mkfs.xfs", []string{"/dev/foo"}, "", fmt.Errorf("formatting failed")},
+			},
+			expectedError: fmt.Errorf("formatting failed"),
+		},
+		{ // Test that 'lsblk' is called and confirms unformatted disk, format passes, second mount fails
+			fstype:    "xfs",
+			mountErrs: []error{fmt.Errorf("unknown filesystem type '(null)'"), fmt.Errorf("Still cannot mount")},
+			execScripts: []ExecArgs{
+				{"lsblk", []string{"-nd", "-o", "FSTYPE", "/dev/foo"}, "", nil},
+				{"mkfs.xfs", []string{"/dev/foo"}, "", nil},
+			},
+			expectedError: fmt.Errorf("Still cannot mount"),
+		},
+		{ // Test that 'lsblk' is called and confirms unformatted disk, format passes, second mount passes
+			fstype:    "xfs",
+			mountErrs: []error{fmt.Errorf("unknown filesystem type '(null)'"), nil},
+			execScripts: []ExecArgs{
+				{"lsblk", []string{"-nd", "-o", "FSTYPE", "/dev/foo"}, "", nil},
+				{"mkfs.xfs", []string{"/dev/foo"}, "", nil},
+			},
+			expectedError: nil,
+		},
```

Comment:
In fact just keep the last test case above and delete the rest. Change to comment to something like "test that none ext4 fs does not get called with ext4 options.